docs/configuration.md

```diff
@@ -0,0 +1,316 @@
+| _observedNodes_        | The number of nodes observed according to the selector.                                                                                                                                                                                                    |
+| _healthyNodes_         | The number of observed healthy nodes.                                                                                                                                                                                                                      |
+| _inFlightRemediations_ | ** DEPRECATED ** A list of "timestamp - node name" pairs of ongoing remediations. Replaced by unhealthyNodes.                                                                                                                                              |
+| _unhealthyNodes_       | A list of unhealthy nodes and their remediations. See details below.                                                                                                                                                                                       |
+| _conditions_           | A list of conditions representing NHC's current state. Currently the only used type is "Disabled", and it is true when the comtroller detects problems which prevent it to work correctly. See the [workflow page](./workflow.md) for further information. |
+| _phase_                | A short human readable representation of NHC's current state. Known phases are Disabled, Paused, Remediating and Enabled.                                                                                                                                  |
+| _reason_               | A longer human readable explanation of the phase.                                                                                                                                                                                                          |
+
+### UnhealthyNodes
+
+TODO add when documenting escalating remediation
+
+## Remediation Resources
+
+There are two kind of remediation resources involved:
+
+- at least one remediation template CR, which needs to exist when configuring
+a NHC CR.
+- the remediation CRs created by NHC for unhealthy nodes, and being processed
+by the remediator providing the related CRDs.
+
+As mentioned above, the NHC CR is referencing one or more remediation templates.
+While NHC does not know the template's CRD, it expects the CR to have a specific
+structure, which looks like this:
+
+```yaml
+apiVersion: remediator.company.io/v1
+kind: MyRemediationTemplate
+metadata:
+  name: test-name
+  namespace: test-namespace
+spec:
+  template:
+    spec: {}
+```
+
+> **Note**
+> 
+> - The kind must have a "Template" suffix.
+> - The content of the inner spec doesn't matter. While it can be empty as in this
```

Comment:
Nit: I know it's implied, but maybe note explicitly that the `spec` must have a `template` field ?